### PR TITLE
[Performance][DSpark] Fuse full-vocabulary greedy Markov reduction for DeepSeek-V4 on Ascend

### PR DESCRIPTION
**Pull request by 2803893700qqcom**
## What this PR does / why we need it?

DSpark greedy drafting selects each token from the sum of backbone logits and the Markov-head bias. This PR adds an opt-in, two-stage Triton reduction for the full-vocabulary DeepSeek-V4 path. It combines source-dtype addition and argmax, then writes the selected token ID directly into the draft-token buffer. The Markov projection remains unchanged.

The fused path is selected only for matching FP16, BF16, or FP32 input dtypes with compatible shapes and vocabulary strides. Probabilistic drafting, top-k drafting, reduced-vocabulary mapping, and incompatible tensor layouts continue through the existing sampler. The temporary reduction buffers are allocated during speculator initialization and reused across draft steps.

Related Upstream Work [local-inference-lab/vllm PR #738](https://github.com/local-inference-lab/vllm/pull/738). 

## Does this PR introduce _any_ user-facing change?

It adds `additional_config.enable_dspark_fused_greedy`, which defaults to `false`. Users must set it to `true` to enable the fused greedy path. When the option is disabled, the existing sampling path is used.

### Performance results

Benchmark setup supplied by the experiment owner:

- Hardware: 4 × Ascend 910B NPUs.
- Target model: DeepSeek-V4-Flash.
- Benchmark tool: AISBench.
- Input length: 32k tokens; output length: 1k tokens.
- Concurrency: 32.
- Comparison: the baseline has `additional_config.enable_dspark_fused_greedy=false`; the optimized run has it set to `true`. All other settings were reported to be identical.

Lower is better for latency; higher is better for throughput. Percentages below are recomputed from the displayed raw values.

| Metric | Baseline | Fused greedy | Change |
| --- | ---: | ---: | ---: |
| E2EL | 68,135.9 ms | 65,828.4 ms | −3.39% |
| TTFT | 50,851.1 ms | 49,211.5 ms | −3.22% |
| TPOT | 16.9 ms | 16.2 ms | −4.14% |
| OutputToken throughput | 16.3709 token/s | 16.9786 token/s | +3.71% |
| Prefill throughput | 646.0245 token/s | 667.5478 token/s | +3.33% |
| Input throughput | 14,397.7338 token/s | 14,890.2086 token/s | +3.42% |
| Output throughput | 448.7917 token/s | 464.1426 token/s | +3.42% |
| Total throughput | 14,846.5254 token/s | 15,354.3512 token/s | +3.42% |

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

File: tests/e2e/pull_request/one_card/spec_decode/test_dspark_greedy.py
Comment (ZT-AIA) [inline tests/e2e/pull_request/one_card/spec_decode/test_dspark_greedy.py:1]: Single-operator test cases should be placed here https://github.com/vllm-project/vllm-ascend/tree/main/tests/e2e/nightly/single_node/ops/singlecard_ops/triton

File: .github/workflows/scripts/estimated_times.yaml [hunk 1]
```diff
@@ -45,6 +45,7 @@ estimated_times:
   tests/e2e/pull_request/one_card/pooling/test_scoring.py: 740
   tests/e2e/pull_request/one_card/spec_decode/test_dflash.py: 510
   tests/e2e/pull_request/one_card/spec_decode/test_dspark.py: 410
+  tests/e2e/pull_request/one_card/spec_decode/test_dspark_greedy.py: 60
```
Comment (ZT-AIA) [inline .github/workflows/scripts/estimated_times.yaml:48]: This modification is not needed.

File: tests/e2e/pull_request/one_card/spec_decode/test_dspark_greedy.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ("vocab", "input_dtype", "output_dtype"),
+    [
+        (1, torch.float32, torch.int32),  # parts=1，验证最小归约块
+        (1, torch.float32, torch.int64),
+        (2048, torch.float16, torch.int64),  # 恰好一个完整分块
+        (2049, torch.float16, torch.int32),  # 跨分块边界
+        (2049, torch.float16, torch.int64),
+        (16385, torch.bfloat16, torch.int32),  # parts=9，补齐到16
+        (16385, torch.bfloat16, torch.int64),
+        (32769, torch.float32, torch.int64),  # parts=17，补齐到32
```
Comment (Tflowers-0129) [inline tests/e2e/pull_request/one_card/spec_decode/test_dspark_greedy.py:22]: 让我们说英文😀

File: vllm_ascend/worker/v2/spec_decode/dspark/greedy.py
Comment (ZT-AIA) [inline vllm_ascend/worker/v2/spec_decode/dspark/greedy.py:1]: Please place the operator implementation here. https://github.com/vllm-project/vllm-ascend/tree/main/vllm_ascend/ops/triton/v2
